diff --git a/kernel/feature/sucompat.c b/kernel/feature/sucompat.c
index aa4a13eb..2f48591d 100644
--- a/kernel/feature/sucompat.c
+++ b/kernel/feature/sucompat.c
@@ -381,7 +381,7 @@ static inline int do_ksu_handle_execveat_sucompat(int *fd, const char *filename,
     }
 
     path_put(&kpath);
-    memcpy((void *)filename, ksud_path, sizeof(ksud_path));
+    memcpy((void *)filename, sh_path, sizeof(sh_path));
 out:
     ksu_sulog_emit_pending(pending_sucompat, 0, GFP_KERNEL);
     // always flag that, to avoid old version of susfs hang in boot
@@ -439,7 +439,11 @@ int ksu_handle_execve(int *fd, const char *filename, void *argv, void *envp, int
 
 #ifndef CONFIG_KSU_TRACEPOINT_HOOK
     if (ksu_is_current_proc_unprivillege()) {
-        return -EINVAL;
+        if (ksu_is_allow_uid_for_current(ksu_get_uid_t(current_uid()))) {
+            ksu_clear_current_proc_unprivillege();
+        } else {
+            return -EINVAL;
+        }
     }
 #endif
 
