diff --git a/kernel/feature/sucompat.c b/kernel/feature/sucompat.c
index aa4a13eb..49a41bd9 100644
--- a/kernel/feature/sucompat.c
+++ b/kernel/feature/sucompat.c
@@ -381,7 +381,7 @@ static inline int do_ksu_handle_execveat_sucompat(int *fd, const char *filename,
     }
 
     path_put(&kpath);
-    memcpy((void *)filename, ksud_path, sizeof(ksud_path));
+    memcpy((void *)filename, sh_path, sizeof(sh_path));
 out:
     ksu_sulog_emit_pending(pending_sucompat, 0, GFP_KERNEL);
     // always flag that, to avoid old version of susfs hang in boot
